### PR TITLE
New userfrontend

### PR DESCRIPTION
**Pull request by ffactory-ofcl**

### CODE REVIEW

File: src/serverMain/kotlin/com/studo/campusqr/endpoints/UserFrontend.kt [hunk 1]
```diff
@@ -126,19 +130,72 @@ suspend fun ApplicationCall.userFrontend() {
...
-              +LocalizedString("Check-in successful!", "Check-in erfolgreich!").get(this@userFrontend)
-            }
           }
-          span("result hidden") {
-            id = "result-fail"
+          span("result fail hidden") {
+            id = "result-not-allowed"
+            +LocalizedString(
+              "Error! You're not allowed to check-in.",
+              "Fehler! Du hast keine Berechtigung um einzuchecken."
```
Comment (vonox7) [inline src/serverMain/kotlin/com/studo/campusqr/endpoints/UserFrontend.kt:194]:
```suggestion
              "Error! You're not allowed to check-in.",
              "Fehler! Sie haben keine Berechtigung um einzuchecken."
```

File: src/serverMain/kotlin/com/studo/campusqr/endpoints/UserFrontend.kt [hunk 2]
```diff
@@ -126,19 +130,72 @@ suspend fun ApplicationCall.userFrontend() {
...
+            ).get(this@userFrontend)
+          }
+          span("result fail hidden") {
+            id = "result-net-err"
             +LocalizedString(
               "Error! Please try again.",
               "Fehler! Bitte versuche es erneut."
```
Comment (vonox7) [inline src/serverMain/kotlin/com/studo/campusqr/endpoints/UserFrontend.kt:201]:
```suggestion
              "Error! Please try again.",
              "Fehler! Bitte versuchen Sie es erneut."
```

File: src/serverMain/resources/userFrontend/userFrontend.css [hunk 3]
```diff
@@ -165,21 +179,173 @@ button#submit-button:active {
...
+.result.fail {
+  text-align: center;
+  margin-top: 32px;
   font-weight: bold;
-  align-self: center;
 }
 
 #result-ok {
-  height: calc(100vh - 64px - 172px - 96px);
+  /*height: calc(100vh - 64px - 172px - 96px);*/
```
Comment (vonox7) [inline src/serverMain/resources/userFrontend/userFrontend.css:198]: ?

File: src/serverMain/resources/userFrontend/userFrontend.css [hunk 4]
```diff
@@ -165,21 +179,173 @@ button#submit-button:active {
...
 }
 
-div.icon-wrap {
+.result > .large {
+  border-radius: 16px;
+  /*box-shadow: 0 0 16px 1px rgba(0, 0, 0, 0.2);
+  border: 2px solid #959EA6;*/
+  width: 100%;
+  padding: 24px;
+  /*margin-bottom: 64px;*/
```
Comment (vonox7) [inline src/serverMain/resources/userFrontend/userFrontend.css:225]: ?

File: src/serverMain/resources/userFrontend/userFrontend.css [hunk 5]
```diff
@@ -165,21 +179,173 @@ button#submit-button:active {
...
+  position: absolute;
+  width: 100%;
+  left: 0;
+  opacity: 0.25;
+  font-weight: bold;
+  font-size: 41px;
+  text-transform: uppercase;
+}
+
+/* https://codepen.io/ClaireLarsen/pen/XmVyVX */
```
Comment (vonox7) [inline src/serverMain/resources/userFrontend/userFrontend.css:273]: ?

File: src/serverMain/resources/userFrontend/userFrontend.css [hunk 6]
```diff
@@ -165,21 +179,173 @@ button#submit-button:active {
...
+}
+
+@keyframes stroke-offset {
+  100% {
+    stroke-dashoffset: -40%;
+  }
+}
+
+/*.result > .large > .number-wrapper > .number {
+}*/
```
Comment (vonox7) [inline src/serverMain/resources/userFrontend/userFrontend.css:322]: ?

File: src/serverMain/resources/userFrontend/userFrontend.css [hunk 7]
```diff
@@ -165,21 +179,173 @@ button#submit-button:active {
...
+  border-radius: 4px;
+  width: 100%;
+  line-height: 2.5;
+  text-align: center;
+  /*box-shadow: 0 2px 8px 1.5px rgba(0, 0, 0, 0.15);*/
+}
+
+/*.result > .details > * {
+  border-bottom: 1px solid #b5b5b5;
+}*/
```
Comment (vonox7) [inline src/serverMain/resources/userFrontend/userFrontend.css]: ?

File: src/serverMain/resources/userFrontend/userFrontend.js [hunk 8]
```diff
@@ -46,39 +73,67 @@ function onLoad() {
...
+              showCheckin(email)
+
+            } else if (this.status === 403) {
+              resultNotAllowed.className = resultNotAllowed.className.replace("hidden", "");
+
+            } else {
+              // Network or internal error
+              resultNetErr.className = resultNetErr.className.replace("hidden", "");
+            }
+          }, 700)
```
Comment (vonox7) [inline src/serverMain/resources/userFrontend/userFrontend.js:100]: Why always waiting?

**Review by vonox7 — CHANGES_REQUESTED** [review]

File: src/serverMain/resources/userFrontend/userFrontend.js (hunk 8 above)
Comment (ffactory-ofcl) [reply src/serverMain/resources/userFrontend/userFrontend.js:100]: Ux / feels nicer than instantly getting the checkin screen.

**Review by vonox7 — APPROVED** [review]

**Pull request merged**